doc/validator-keys-tool-guide.md

```diff
@@ -0,0 +1,99 @@
+
+## Master Keys
+
+When first setting up a validator, use the `validator-keys` tool to generate a
+master key pair:
+
+```
+  $ validator-keys create_master_keys
+```
+
+Sample output:
+```
+  Master validator keys stored in /home/ubuntu/.ripple/validator-master-keys.json
+```
+
+Sample key file:
+```
+  {
+     "key_type" : "ed25519",
+     "master_public_key" : "nHUtNnLVx7odrz5dnfb2xpIgbEeJPbzJWfdicSkGyVw1eE5GpjQr",
+     "master_secret_key" : "paUvjGIz51cpSxL3Wc9DaGTEogYk4AEesLJJaawBcVQjejxGFcb",
+     "sequence" : 1
+  }
+```
+
+The `master_public_key` is the master public key. Any other rippled trusting
+the validator needs to add the master public key to the [validators] section
+of its config. Only add keys received from trusted sources.
+
+The `master_secret_key` is the corresponding master secret key. **DO NOT SHARE
+THIS KEY**. The master secret key will be used to sign manifests that change
+ephemeral keys. Put the master secret key in a secure but recoverable location.
+
+## Ephemeral Keys
+
+After first creating the [master keys](#master-keys) or when changing the
+ephemeral keys, use the `validator-keys` tool to create a new ephemeral key pair:
+
+```
+  $ validator-keys create_ephemeral_keys
+```
+
+Sample output:
+
+```
+  Update rippled.cfg file with these values:
+
+  [validation_seed]
+  shDaxelXa4EKNt65ZxXUZBRaF2aE2
+  # validation_public_key: n9LSS563nUSjgKo49CmMfnHM6tBerUlMLvL7o1GMwvTwXi5UmpWT
+  # sequence number: 1
+
+  [validation_manifest]
+  JAAAAAFxIe3Jmw2YP06xDcJ/9k7gBjByXwoduRwMZcXH578DGNk1m3MhAvmlIKVGDqdyCU/6
+  HIubKFtFhVQefrRr7SLhN48BP5nMdkcwRQIhAI4fn64/8XgJeuVR1Eu6lu50XBC2CcPPAUSf
+  Ta7KizeYAiAwpjaahPWW9muj2FA/mAB/zJbrMKL3c462r97JTvi1FHASQDNj/jMpCrZl77r4
+  zmeE8pY9cZC5aS+3Jk6y2wvaSG7Szq7Mfc8Hcpm7gBmSVbbQ+rl+DGBgaVnkkDscXvdOpQ4=
+
+```
+
+Add the `[validation_seed]` and `[validation_manifest]` values to this
+validator's config and restart rippled.
+
+The manifest is a signed message used to inform other servers of this
+validator's ephemeral public key. A manifest contains a sequence number,
+the master public key, and the new ephemeral validation public key, and
+it is signed with both the ephemeral and master secret keys.
+The sequence number should be higher than the previous sequence number (if it
+is not, the manifest will be ignored). The `validator-keys` tool will
+automatically increment the sequence number.
+
+## Revoking a key
+
+If a master key is compromised, the key may be revoked permanently. To revoke
+the master key, use the `validator-keys` tool to sign a manifest with the
+highest possible sequence number (4294967295), which is a special value to
+indicate to other servers that the master key is no longer valid:
+
+```
+  $ validator-keys revoke_master_keys
+```
+
+Add the `[validation_seed]` and `[validation_manifest]` values to this
+validator's config and restart rippled. After rippled has run for at least a
+few minutes, rename the old key file and generate new [master](#master-keys) and
```

Comment:
I'm concerned about the phrase, "after rippled has run for at least a few minutes".

You should let them know what they are accomplishing by running their validator this way.  They need to run _synchronized_ to the network for at least a few minutes, right?  Is there a way they can tell the the revocation has spread sufficiently across the network?

To me the "at least a few minutes" guidance is not sufficient.  If I'm an ops person just trying to get something over and done with, running rippled for just a few minutes may not even get the server synced.  They may not accomplish the goal.  If I'm a person worried about really revoking the old key then saying "at least a few minutes" does not give me a feeling for how many minutes will be sufficient (and why).  In one sense the more minutes they do this the more minutes their validator is off line.  In the other sense they want to run this way long enough to make sure the key is really revoked on the network.